### ISSUE TITLE
custom types are no longer used to ID model since v0.17.24

### ISSUE THREAD

**Issue by shirakiya** [99designs/gqlgen#2561]
### What happened?
(Please refer to `Minimal graphql.schema and models to reproduce` section.)

Since v0.17.24, I get a type error in the generated code (`generated.go`) when I add a custom type to an ID model in models. v0.17.22 or earlier, `generated.go` would do the following type conversion inside. In contrast with v0.17.24, `graphql.MarshalID`, `graphql.UnmarshalID` is applied to my custom type in the generated code, resulting in a type error.

a part of `generated.go` built with v0.17.22

```golang
func (ec *executionContext) unmarshalNID2githubᚗcomᚋshirakiyaᚋgqlgenᚑtestᚋgraphᚋmy_packageᚐUUID(ctx context.Context, v interface{}) (my_package.UUID, error) {
	tmp, err := graphql.UnmarshalString(v)
	res := my_package.UUID(tmp)
	return res, graphql.ErrorOnPath(ctx, err)
}

func (ec *executionContext) marshalNID2githubᚗcomᚋshirakiyaᚋgqlgenᚑtestᚋgraphᚋmy_packageᚐUUID(ctx context.Context, sel ast.SelectionSet, v my_package.UUID) graphql.Marshaler {
	res := graphql.MarshalString(string(v))
	if res == graphql.Null {
		if !graphql.HasFieldError(ctx, graphql.GetFieldContext(ctx)) {
			ec.Errorf(ctx, "the requested element is null which the schema does not allow")
		}
	}
	return res
}
```

a part of `generated.go` built with v0.17.24

```golang
func (ec *executionContext) unmarshalNID2githubᚗcomᚋshirakiyaᚋgqlgenᚑtestᚋgraphᚋmy_packageᚐUUID(ctx context.Context, v interface{}) (my_package.UUID, error) {
	res, err := graphql.UnmarshalID(v)  // Applied `graphql.UnmarshalID()` against my will. So type error occurs.
	return res, graphql.ErrorOnPath(ctx, err)
}

func (ec *executionContext) marshalNID2githubᚗcomᚋshirakiyaᚋgqlgenᚑtestᚋgraphᚋmy_packageᚐUUID(ctx context.Context, sel ast.SelectionSet, v my_package.UUID) graphql.Marshaler {
	res := graphql.MarshalID(v)
	if res == graphql.Null {
		if !graphql.HasFieldError(ctx, graphql.GetFieldContext(ctx)) {
			ec.Errorf(ctx, "the requested element is null which the schema does not allow")
		}
	}
	return res
}
```

with following error when gqlgen runs

```
go generate ./...
validation failed: packages.Load: /Users/shirakiya/sandbox/gqlgen-test/graph/generated.go:3166:9: cannot use res (variable of type string) as my_package.UUID value in return statement
/Users/shirakiya/sandbox/gqlgen-test/graph/generated.go:3170:27: cannot use v (variable of type my_package.UUID) as string value in argument to graphql.MarshalID
```

### What did you expect?
I expect code generation to succeed even if custom types are defined for ID model as was the case prior to v0.17.22.

As a side note, I understand the following type definition prevents type errors. But my business codes disallow the definition because it requires too large changes, so it's unrealistic solution x(

```golang
type UUID = string
```

### Minimal graphql.schema and models to reproduce
After following the steps at https://gqlgen.com/getting-started/, I made the following changes. 

schema.graqhqls

```
type Todo {
  id: ID!
  text: String!
  done: Boolean!
  user: User!
}

type User {
  id: ID!
  name: String!
}

type Query {
  todos: [Todo!]!
}

input NewTodo {
  text: String!
  userId: ID!  # change this line compared with the official guide
}

type Mutation {
  createTodo(input: NewTodo!): Todo!
}
```

gqlgen.yml

```yaml
models:
  ID:
    model:
      - github.com/99designs/gqlgen/graphql.ID
      - github.com/99designs/gqlgen/graphql.Int
      - github.com/99designs/gqlgen/graphql.Int64
      - github.com/99designs/gqlgen/graphql.Int32
      - github.com/shirakiya/gqlgen-test/graph/my_package.UUID
  Int:
    model:
      - github.com/99designs/gqlgen/graphql.Int
      - github.com/99designs/gqlgen/graphql.Int64
      - github.com/99designs/gqlgen/graphql.Int32
  Todo:
    model: github.com/shirakiya/gqlgen-test/graph/my_package.Todo
    fields:
      user:
        resolver: true
  User:
    model: github.com/shirakiya/gqlgen-test/graph/my_package.User
```

<root>/graph/my_package/model.go

```golang
package my_package

type Todo struct {
	ID     string `json:"id"`
	Text   string `json:"text"`
	Done   bool   `json:"done"`
	UserID UUID   `json:"userId"`
	User   *User  `json:"user"`
}

type UUID string

type User struct {
	ID   UUID   `json:"id"`
	Name string `json:"name"`
}
```

### versions
 - `v0.17.24`
 - `go version go1.19.5 darwin/arm64`

**Comment by mstephano** [comment]
Hi, I will investigate your issue and will get back to you. Sorry about this!

**Comment by jogly** [comment]
I found this report while investigating a very similar break in my project.  My project uses `uint` for the scalar `ID`, and `v0.17.22` generated with no errors, while `v0.17.23` generates code with an almost identical error to @shirakiya 's:

`validation failed: packages.Load: ...... : cannot use &res (value of type *int64) as *uint value in return statement`. The error is the same in `v0.17.24`.

**Comment by mstephano** [comment]
Sorry been very busy at work lately and I should be able to verify your issue soon.

**Comment by FlymeDllVa** [comment]
I confirm that scalar mapping has stopped working correctly :(

**Comment by mdestagnol** [comment]
same here

**Comment by StevenACoffman** [comment]
Please see these recent changes which are related:
#2528
#2468
#2486
#2508
#2528

**Comment by StevenACoffman** [comment]
It doesn't seem like @mstephano is able to devote time to this, so I would be very interested in anyone's PR that would fix this while avoiding regressions.

**Comment by mihaimaruseac** [comment]
Confirming this is an issue for us too. ID fields suddenly translate to `string`, even though before they were translating to integers

**Comment by StevenACoffman** [comment]
See if this is fixed with v0.17.27 and let me know!

**Comment by shirakiya** [comment]
It's fixed with v0.17.27. Thanks for your work!
